api/controllers/productController.js

```diff
@@ -0,0 +1,19 @@
+const { productService } = require("../services");
+const { catchAsync } = require("../utils/error");
+
+const productList = catchAsync(async (req, res) => {
+  const { categoryId, size, color } = req.query;
+  try {
+    const list = await productService.productList(categoryId, size, color);
+    res.status(200).json({
+      data: list,
+    });
+  } catch (err) {
+    console.log(err);
+    return res.status(err.statusCode || 500).json({ message: err.message });
+  }
+});
```

Comment:
catchAsync를 활용하고 있기 때문에 지금의 코드에서는 기본적으로 try catch 문이 필요하지 않습니다.
```suggestion
const productList = catchAsync(async (req, res) => {
  const { categoryId, size, color } = req.query;
  
  const list = await productService.productList(categoryId, size, color);
  
  return res.status(200).json({
    data: list,
  });
});
```